src/VisualStudio/Core/Def/SolutionEventMonitor.cs

```diff
@@ -45,6 +45,16 @@ public SolutionEventMonitor(VisualStudioWorkspace workspace)
                 }
 
                 KnownUIContexts.SolutionOpeningContext.UIContextChanged += SolutionOpeningContextChanged;
+
+                _solutionClosingContext = UIContext.FromUIContextGuid(VSConstants.UICONTEXT.SolutionClosing_guid);
```

Comment:
💡 Move this initialization outside the `if` statement

Reply:
@sharwell Entire functionality within the `SolutionEventMonitor` type is conditioned on the below if check inside its constructor: https://github.com/dotnet/roslyn/blob/fa9926dbc9d6b6677f96efc5b47e72037562ae1b/src/VisualStudio/Core/Def/SolutionEventMonitor.cs#L28

I am just adding to existing patterns for solution open and build contexts, so am hesitant to introduce inconsistency in the code on how solution close is dealt as compared to open/build.

Reply:
@sharwell If you still feel strongly about this, a separate PR cleaning up this entire type and possibly removing the below check would be more suitable: https://github.com/dotnet/roslyn/blob/fa9926dbc9d6b6677f96efc5b47e72037562ae1b/src/VisualStudio/Core/Def/SolutionEventMonitor.cs#L28

Reply:
Filed https://github.com/dotnet/roslyn/issues/58827 if we want to cleanup this type.